test/Microsoft.AspNetCore.Sockets.Common.Tests/Internal/Formatters/ServerSentEventsParserTests.cs

```diff
@@ -0,0 +1,87 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.IO;
+using System.Text;
+using Microsoft.AspNetCore.Sockets.Internal.Formatters;
+using Xunit;
+using System.Threading.Tasks;
+using System.IO.Pipelines;
+using System;
+
+namespace Microsoft.AspNetCore.Sockets.Common.Tests.Internal.Formatters
+{
+    public class ServerSentEventsParserTests
+    {
+        [Theory]
+        [InlineData("data: T\r\ndata: Hello, World\r\n\r\n", MessageType.Text, "Hello, World")]
+        [InlineData("data: E\r\ndata: Hello, World\r\n\r\n", MessageType.Error, "Hello, World")]
+        [InlineData("data: T\r\ndata: Hello\r\ndata: , World\r\n\r\n", MessageType.Text, "Hello, World")]
+        [InlineData("data: T\r\ndata: Major\r\ndata:  Key\r\ndata:  Alert\r\n\r\n", MessageType.Text, "Major Key Alert")]
+        [InlineData("data: T\r\n\r\n", MessageType.Text, "")]
+        public async Task ParseSSEMessageSuccessCases(string encodedMessage, MessageType messageType, string expectedMessage)
+        {
+            var stream = new MemoryStream();
+            var streamWriter = new StreamWriter(stream);
+            streamWriter.Write(encodedMessage);
+            streamWriter.Flush();
+            stream.Seek(0, SeekOrigin.Begin);
+
+            var pipelineReader = stream.AsPipelineReader();
+            var buffer = await pipelineReader.ReadToEndAsync();
```

Comment:
Rather than using a real pipeline, you can just create a `ReadableBuffer` directly from an array using `ReadableBuffer.Create` and save some steps. See https://github.com/aspnet/SignalR/blob/dev/test/Microsoft.AspNetCore.Sockets.Common.Tests/Internal/Formatters/TextMessageParserTests.cs#L27 (which uses `BytesReader`, which is a different thing, but the pattern should be about the same for `ReadableBuffer`)